[BUG] simulation CancelTask error with info.canceled field

**Describe the bug**
Simulation error with `govmomi.simulator.task.go` CancelTask() when testing in pipelines, specifically Docker images golang:1.21 & golang:1.22, although it's possible this issue is more prevalent than just inside containers.

Error lies with the property field `info.canceled` on line 260 being invalid to the http server, causing a panic. I found a fix to be changing the field to **info.cancelled** instead.

Here's a minimal Go test that I wrote that shows this error when trying to update the `info.canceled` field of a mock task: 

```
import (
	"context"
	"testing"

	"github.com/go-playground/assert/v2"
	"github.com/vmware/govmomi/simulator"
	"github.com/vmware/govmomi/vim25"
	"github.com/vmware/govmomi/vim25/mo"
	"github.com/vmware/govmomi/vim25/types"
)

type fakeTask struct {
	*mo.Folder

	fault types.BaseMethodFault
}

func (t *fakeTask) Run(task *simulator.Task) (types.AnyType, types.BaseMethodFault) {
	return nil, t.fault
}

func TestCancelTask(t *testing.T) {
     simulator.Run(func(ctx context.Context, c *vim25.Client) error {
     
            folder := &mo.Folder{}
            simulator.Map.NewEntity(folder)
            
            t.Run("info.canceled bug", func(t *testing.T) {
			task := simulator.NewTask(&fakeTask{folder, nil})
			task.RunBlocking(simulator.SpoofContext())
			taskInfo := &task.Info
			taskInfo.State = types.TaskInfoStateQueued

			changes := []types.PropertyChange{
				{Name: "info.canceled", Val: true},}
		
			simulator.Map.Update(task, changes) // panic happens here
	     })
	     
	     return nil
    })
}

```

Interestingly, I've found this bug is only occurring in golang containers, not on my machine itself. However, it does not occur on the containers when the field is set to **info.cancelled**, and it also does not occur on my machine.

Since `info.canceled` is the name of the field in `govmomi/simulator/task.go` CancelTask() line 260, that method fails when being tested against with the same http panic error on line 269.

**To Reproduce**
Steps to reproduce the behavior:

1. Run the minimal Go test in a Docker image for golang:1.21 or 1.22
2. The test should fail with an http panic error
3. Change field to **info.cancelled** and repeat steps 1 and test will pass

**Expected behavior**
The expected behavior of CancelTask() in the simulation package would be to successfully update the fields of the Task for **info.cancelled**, and to not have an http panic error.

**Affected version**
Affects govmomi versions 0.38.0 & 0.39.0

**Screenshots/Debug Output**
Here's what happens when I try to do a CancelTask with this test (using my one liner below):

```
import (
	"context"
	"testing"

	"github.com/vmware/govmomi/simulator"
	"github.com/vmware/govmomi/vim25"
	"github.com/vmware/govmomi/vim25/methods"
	"github.com/vmware/govmomi/vim25/mo"
	"github.com/vmware/govmomi/vim25/types"
)

type fakeTask struct {
	*mo.Folder

	fault types.BaseMethodFault
}

func (t *fakeTask) Run(task *simulator.Task) (types.AnyType, types.BaseMethodFault) {
	return nil, t.fault
}

func TestRealCancelTask(t *testing.T) {
	simulator.Run(func(ctx context.Context, c *vim25.Client) error {

		folder := &mo.Folder{}
		simulator.Map.NewEntity(folder)

		task := simulator.NewTask(&fakeTask{folder, nil})
		task.RunBlocking(simulator.SpoofContext())
		taskInfo := &task.Info
		taskInfo.State = types.TaskInfoStateQueued

		_, err := methods.CancelTask(ctx, c, &types.CancelTask{
			This: taskInfo.Task,
		})
		if err != nil {
			t.Fatal(err)
		}

		return nil
	})
}
```

```
2024/07/26 23:12:07 http: panic serving 127.0.0.1:56102: info.canceled: property not found
goroutine 16 [running]:
net/http.(*conn).serve.func1()
	/usr/local/go/src/net/http/server.go:1873 +0xb9
panic({0xd05cc0?, 0xc00050a890?})
	/usr/local/go/src/runtime/panic.go:920 +0x270
github.com/vmware/govmomi/vim25/mo.ApplyPropertyChange({0x116e460, 0xc000476000}, {0xc000000b40, 0x4, 0xd71680?})
	/go/pkg/mod/github.com/vmware/govmomi@v0.39.0/vim25/mo/retrieve.go:85 +0x334
github.com/vmware/govmomi/simulator.(*Registry).Update(0xfd4680?, {0x116e000, 0xc000476000}, {0xc000000b40, 0x4, 0x4})
	/go/pkg/mod/github.com/vmware/govmomi@v0.39.0/simulator/registry.go:321 +0x20d
github.com/vmware/govmomi/simulator.(*Task).CancelTask(0xc000476000, 0xc000738480, 0x1?)
	/go/pkg/mod/github.com/vmware/govmomi@v0.39.0/simulator/task.go:269 +0x330
reflect.Value.call({0xf77e60?, 0xc000476000?, 0x7fc90ca48f18?}, {0xfe1ed2, 0x4}, {0xc000735230, 0x2, 0xc00014d2e0?})
	/usr/local/go/src/reflect/value.go:596 +0xce7
reflect.Value.Call({0xf77e60?, 0xc000476000?, 0x30?}, {0xc000735230?, 0xc000799aa0?, 0x46ecd3?})
	/usr/local/go/src/reflect/value.go:380 +0xb9
github.com/vmware/govmomi/simulator.(*Service).call.func1()
	/go/pkg/mod/github.com/vmware/govmomi@v0.39.0/simulator/simulator.go:217 +0x34
github.com/vmware/govmomi/simulator.(*Registry).WithLock(0xc0005a82a0?, 0xc000476000?, {0x116e000?, 0xc000476000?}, 0xc00014d610)
	/go/pkg/mod/github.com/vmware/govmomi@v0.39.0/simulator/registry.go:649 +0x34
github.com/vmware/govmomi/simulator.(*Service).call(0xc00008a5a0, 0xc000738480, 0xc0007385a0)
	/go/pkg/mod/github.com/vmware/govmomi@v0.39.0/simulator/simulator.go:216 +0xf08
github.com/vmware/govmomi/simulator.(*Service).ServeSDK(0xc00008a5a0, {0x1173eb8, 0xc0004467e0}, 0xc00042a300)
	/go/pkg/mod/github.com/vmware/govmomi@v0.39.0/simulator/simulator.go:501 +0x586
net/http.HandlerFunc.ServeHTTP(0x10?, {0x1173eb8?, 0xc0004467e0?}, 0xc00019aa2c?)
	/usr/local/go/src/net/http/server.go:2141 +0x29
net/http.(*ServeMux).ServeHTTP(0x412ae5?, {0x1173eb8, 0xc0004467e0}, 0xc00042a300)
	/usr/local/go/src/net/http/server.go:2519 +0x142
net/http.serverHandler.ServeHTTP({0x1172388?}, {0x1173eb8?, 0xc0004467e0?}, 0x6?)
	/usr/local/go/src/net/http/server.go:2943 +0x8e
net/http.(*conn).serve(0xc00079a5a0, {0x1174550, 0xc000810660})
	/usr/local/go/src/net/http/server.go:2014 +0x5f4
created by net/http.(*Server).Serve in goroutine 82
	/usr/local/go/src/net/http/server.go:3091 +0x5cb
--- FAIL: TestRealCancelTask (0.43s)
    task_manager_test.go:97: Post "https://127.0.0.1:33991/sdk": EOF
```

**Additional context**
You can use this one liner to quickly test:
`docker run --rm -v $(pwd):/go/src/app -w /go/src/app golang:1.21 go test -v {test_file}`

--- a/govc/task/create.go
+++ b/govc/task/create.go
@@ -94,7 +94,7 @@ func (cmd *create) Run(ctx context.Context, f *flag.FlagSet) error {
 		return err
 	}
 
-	fmt.Println(res.Returnval.Task)
+	fmt.Println(res.Returnval.Task.Value)
 
 	return nil
 }

--- a/govc/test/tasks.bats
+++ b/govc/test/tasks.bats
@@ -56,6 +56,15 @@ load test_helper
   assert_success
   task="$output"
 
+  run govc task.cancel "$task"
+  assert_success
+  cancelled=$(govc tasks -json | jq ".tasks[] | select(.key == \"$task\") | .cancelled")
+  assert_equal "true" "$cancelled"
+
+  run govc task.create "$id"
+  assert_success
+  task="$output"
+
   run govc task.set -s running "$task"
   assert_success
 

--- a/simulator/task.go
+++ b/simulator/task.go
@@ -257,7 +257,7 @@ func (t *Task) CancelTask(ctx *Context, req *types.CancelTask) soap.HasFault {
 	}
 
 	changes := []types.PropertyChange{
-		{Name: "info.canceled", Val: true},
+		{Name: "info.cancelled", Val: true},
 		{Name: "info.completeTime", Val: time.Now()},
 		{Name: "info.state", Val: types.TaskInfoStateError},
 		{Name: "info.error", Val: &types.LocalizedMethodFault{

